`dotnet new` updates for .NET 7 (#31943)
dotnet/docs: docs/core/tools/dotnet-new-sdk-templates.md

The templates shipped with .NET SDK have the following additional options:

## `console`

- **`-f|--framework <FRAMEWORK>`**

  Specifies the [framework](../../standard/frameworks.md) to target. Available since .NET Core 3.0 SDK.

  The following table lists the default values according to the SDK version you're using:

  | SDK version | Default value   |
  |-------------|-----------------|
  | 7.0         | `net7.0`        |
  | 6.0         | `net6.0`        |
  | 3.1         | `netcoreapp3.1` |

  The ability to create a project for an earlier TFM depends on having that version of the SDK installed. For example, if you have only the .NET SDK 7.0.x installed, then the only value available for `--framework` is `net7.0`. If you install the .NET SDK 6.0.x, the value `net6.0` becomes available for `--framework`. If you install the .NET Core 3.1 SDK, `netcoreapp3.1` becomes available, and so on. So by specifying `--framework netcoreapp3.1` you can target .NET Core 3.1 even while running `dotnet new` in the .NET SDK 6.0.x.

> 

Author: @tdykstra could you please advice: if we want to say that smth is applicable for the whole major version, what is the better way:
- .NET 6 SDK / .NET 7 SDK
- .NET SDK 6.0.x / .NET SDK 7.0.x?

> 

Reviewer: .NET 6 SDK / .NET 7 SDK

- **`--use-program-main`**

>   If specified, an explicit `Program` class and `Main` method will be used instead of top-level statements. Available since .NET SDK 6.0.300. Default value: `false`. Available only in C# language.

Reviewer: ```suggestion
If specified, an explicit `Program` class and `Main` method will be used instead of top-level statements. Available since .NET SDK 6.0.300. Default value: `false`. Available only for C#.
```
